Detectors/util/ecal_layer_stack.py

```diff
@@ -113,52 +309,128 @@ def silicon() :
     def carbon(t) :
         return Layer('Carbon',t)
 
-    def enumerate_full_stack(sections) :
+
+@dataclass
+class BiLayerSandwich:
+    """A "bilayer" in the ECal consists of two (hence 'bi-') sensitive
+    silicon layers mounted onto a central carbon cooling plane.
+
+    There are additional layers of air, PCBs, and potentially absorber.
+    The absorber layers are separated into two areas:
+    - "cooling": these absrobers are mounted between the carbon cooling plane
+        and the hexaboards
+    - "front": these absorbers are put "in front" (upstream/lower z) of the
+        bilayer so that there is absorber between the downstream sensitive
+        hexamodule of the upstream bilayer and the upstream sensitive hexamodule
+        of this bilayer. For many bi-layers, the "front" absorber is twice the
+        thickness of the "cooling" so that approximately the same absorber is
+        between the sensitive hexamodules. In the real detector, the "front" absorber
+        will actually be just two copies of the "cooling" absorber plates in
+        this case.
+
+    The full implementation is given below, but a diagram might be helpful.
+
+       ↓ beam ↓ going down
+
+      | front absorber (W)            |
+      | Readout Motherboard (PCB)     |
+      | Air                           |
+      | Hexamodule (Si, Glue, PCB, C) |
+      | Cooling Absorber (W)          |
+      | Carbon Cooling Plane (Carbon) |
+      | Cooling Absorber (W)          |
+      | Hexaboard (Si, Glue, PCB)     |
+      | Air                           |
+      | Readout Motherboard (PCB)     |
+
+    """
+
+    front: float = 0.0
+    cooling: float = 0.0
```

Comment:
The only thing I'd request to add is that these `front ` and `cooling` are meant to be understood in "mm"

Reply:
probably as "Attributes" similar like you did for `PDGMaterial` class too